frontend/packages/ux-editor/src/containers/FormDesigner.tsx

```diff
@@ -98,6 +114,19 @@ export const FormDesigner = ({ selectedLayout, selectedLayoutSet }: FormDesigner
     const mappedError = mapErrorToDisplayError();
     return <ErrorPage title={mappedError.title} message={mappedError.message}/>;
   }
+  
+  const setSelectedLayoutInLocalStorage = (layoutName: string ) => {
+    if (instanceId) {
+      // Need to use InstanceId as storage key since apps uses it and it is needed to sync layout between preview and editor
+      localStorage.setItem(instanceId, layoutName);
+    }
+  };
+
+  const getSelectedLayoutInLocalStorage = (): string => {
+    if (instanceId) {
+      return localStorage.getItem(instanceId);
+    }
+  };
```

Comment:
I suggest moving these functions on top of the file, outside of the FormDesigner component. This should remove the `dependency array` typecheck warning

```suggestion
  
  const setSelectedLayoutInLocalStorage = (instanceId: string, layoutName: string) => {
    if (instanceId) {
      // Need to use InstanceId as storage key since apps uses it and it is needed to sync layout between preview and editor
      localStorage.setItem(instanceId, layoutName);
    }
  };

  const getSelectedLayoutInLocalStorage = (instanceId: string): string => {
    if (instanceId) {
      return localStorage.getItem(instanceId);
    }
  };
```